hoomd/mpcd/update.py

```diff
@@ -0,0 +1,112 @@
+
+    "Craig M. Tenney and Edward J. Maginn. Limitations and recommendations for
+    the calculation of shear viscosity using reverse nonequilibrium molecular
+    dynamics. J. Chem. Phys. 132, 014103, 2010."
+    `<https://doi.org/10.1063/1.3276454>`_
+
+    "Florian Mueller-Plathe. Reversing the perturbation in nonequilibrium
+     molecular dynamics: An easy way to calculate the shear viscosity of
+     fluids. Phys. Rev. E, 59:4894-4898, May 1999."
+     `<https://doi.org/10.1103/PhysRevE.59.4894>`_
+
+    The simulation box is divided into a number of slabs in y-direction. Two
+    exchange slabs separated by a distance of Ly/2 are chosen (typically the
+    bottom and middle slabs for convenience). Particles are sorted into the
+    slabs based on their positions and candidate particles, in the two exchange
+    slabs, with momentum close to a specified target are selected for a
+    pairwise momentum exchange (of their x-components). In the lower slab,
+    particles with positive momentum in flow direction are selected while those
+    with negative momentum are selected in the upper slab.
+
+    To simulate the orginal implementation by
+    `Mueller-Plathe <https://doi.org/10.1103/PhysRevE.59.4894>`_, only the
+    fastest and slowest particles in the bottom and upper slabs, respectively,
+    should be swapped. This can be acieved by setting the `num_swaps` argument
+    to 1 and the `target_momentum` argument to a sufficiently large value.
+
+    This swapping of particle momenta imposes a momentum flux on the sytstem,
+    which drives the flow. The strength of the flow is set through the
+    `target_momentum` argument, which defines a target value for the
+    time-integrated momentum flux. The searching and swapping is repeated until
+    the target is reached.
+
+    Args:
+        trigger (hoomd.trigger.trigger_like): Trigger to swap momentum.
+
+        num_swaps (int): Maximum number of particle pairs to swap.
+
+        slab_wdith (float): Thickness of momentum-exchange slabs.
+
+        target_momentum (float): Target momentum for swapped particles.
+
+    .. rubric:: Example:
+
+    .. code-block:: python
+
+            flow = hoomd.mpcd.update.ReverseNonequilibriumShearFlow(
+                trigger=20, num_swaps=10, slab_width=0.25, target_momentum=5.0)
+            simulation.operations.updaters.append(flow)
+
+    Attributes:
+        trigger (hoomd.trigger.trigger_like): Trigger to swap momentum.
+
+        num_swaps (int): Maximum number of particle pairs to swap.
+
+        slab_wdith (float): Thickness of momentum-exchange slabs.
+
+        target_momentum (float): Target momentum for swapped particles.
+
+        summed_exchanged_momentum (float): total exchanged momentum between
+            all swapping particles till the current timestep (*read only*)
+
+    """
+
+    # Constructor
+    def __init__(self, trigger, num_swaps, slab_width, target_momentum):
```

Comment:
We should default this to `None`, then if it's None here replace it by the maximum size of a 32-bit or 64-bit float before filling the parameter dictionary (depending on how HOOMD was compiled).
```suggestion
    def __init__(self, trigger, num_swaps, slab_width, target_momentum=None):
```

Reply:
I was thinking about this, and I think this won’t work as proposed due to underflow. I have another idea, though, so please message me when you’re ready to do this part!